suite/tests/runall.cmake

```diff
@@ -168,7 +168,9 @@ endwhile ()
 file(READ "${out}" output)
 # we require that all runall tests write at least one line up front
 set(iters 0)
-if (NOT "${nudge}" MATCHES "<attach>")
+if ("${nudge}" MATCHES "<attach>")
+  do_sleep(0.5)
```

Comment:
If we need a real synchronization point, better to look for a sign like the loop below than just waiting and hoping the timing works out which is a recipe for flakiness.

Reply:
I actually don't think we need it any synchronization at this point - we only need to know that the process is up, and we verify that using the pid file.

I am just trying to understand why the test passes on my machine, but fails in the CI.

Reply:
> I think I fixed it. The issue was that because we do not verify the first line, when we verify "more input" it would be the first line, and its possible that the exit event was not registered yet, so we would timeout waiting for "done".

What was your fix for this?  The code seems to still not wait for a first line.

Reply:
Or is this the "init thread" wait?

Reply:
The output file now has the "Thank you for testing..." line so could this first line wait code be re-instated for attach?  (Maybe I"m mis-remembering what was there before...was that there before?)